src/floor/RequestSimulator.java

```diff
@@ -24,35 +27,73 @@ public RequestSimulator(int f) {
 		}
 
 		floorNum = f;
+		time = 0;
 	}
 
 	@Override
 	public void run() {
+		ElevatorRequestMessage m;
+		FloorMetaMessage f;
+		
+		try (BufferedReader br = new BufferedReader(new FileReader(SimulationVars.inputFile))) {
+			String line;
+			while ((line = br.readLine()) != null) {
+				String[] parts = line.split(" ");
 
-		try {
-			Thread.sleep(2000);
-		} catch (InterruptedException e) {
+				//our next message
+				if (Integer.parseInt(parts[1]) == floorNum) {
```

Comment:
Could you set the `parts[x]` as variables? It was a bit hard to read without referring back to the input file format each time